Automattic/jetpack
# Gutenpack Subscription Block

This pull request adds the Gutenberg Subscription block.

The Subscription block invites visitors to sign up to be notified when there are new posts on a site.

When the block has focus in the editor, editors can toggle whether to show the number of subscribers in the block. When it doesn't have focus, editors will see the number of subscribers if they toggled that on.

#### Testing instructions:
Note that the block still is going to receive some CSS ❤️(cc @MichaelArestad) , so please focus on testing its functionality for now.

1) Checkout this Jetpack branch and the corresponding Calypso branch, 
https://github.com/Automattic/wp-calypso/pull/28887

2) Run the command to build the Subscription block from your wp-calypso repo directory:

```
npm run sdk gutenberg client/gutenberg/extensions/presets/jetpack -- --output-dir=/<path_to_jetpack_repo>/_inc/blocks -w
```
3) Connect Jetpack if it isn't connected

4) Create a new post and add the Jetpack Subscription Block.

5) Toggle the "show number of subscribers". Confirm that you see the number of subscribers when the block loses focus.

6) Publish the post. Confirm that you see it on the frontend. 

7) Subscribe by using the block with a user or two (they will have to be a connected Jetpack user)

8) Confirm that you see the number of subscribed users on the frontend (when the toggle was enabled for the block in the editor) and the editor when the block doesn't have focus.

#### Proposed changelog entry for your changes:

No changelog entry needed.
